config/api.yml

```diff
@@ -540,6 +540,19 @@
         :identifier: instance_guest_restart
       - :name: reset
         :identifier: instance_reset
+  :notifications:
+    :description: User's past notifications
+    :options:
+    - :collection
+    - :arbitrary_resource_path
+    :verbs: *gp
+    :klass: NotificationRecipient
+    :collection_actions:
+      :post:
+      - :name: mark_as_seen
```

Comment:
No need to authorize user. If the user is recipient he has the right. When `@auth_user_obj.notification_recipients.find(id)` it is for that user only.

The DELETE action may come in future to the API. I wanted to introduce notification in smaller iterations. As ui team depends on this work and available time is shortening.

Based on the requirements, the notifications will age/fade with time, so we will need background job to get a rid of past/aged notifications.